src/linalg/impl_linalg.rs

```diff
@@ -454,6 +462,55 @@ fn mat_mul_impl<A>(
         }
         gemm!(f32, cblas_sgemm);
         gemm!(f64, cblas_dgemm);
+
+        macro_rules! gemm {
```

Comment:
Is it possible to share this code with the previous gemm macro? It's hair enough as it is, we don't want to duplicate that code.

Reply:
Yeah, the only difference is that with `Complex<T>` we need to cast through the pointer, whereas `cast_as` doesn't work. I can probably make a shared macro that generates the preamble to be re-used by each macro I suppose.